Sources/SocksCore/Error.swift

```diff
@@ -26,7 +26,7 @@ public enum ErrorReason {
     case localAddressResolutionFailed
     case remoteAddressResolutionFailed
     case ipAddressResolutionFailed
-    case ipAddressValidationFailed
+    case ipAddressValidationFailed(String)
```

Comment:
@andreasley unfortunately, this would break public api and force us to a 2.0 which we're not quite ready for yet. Is it reasonable for us to sacrifice the associated value and perhaps log it where we'd normally set it?